### PR TITLE
fix(uploads): Fix hanging on resuming uploads

### PR DESCRIPTION
**Pull request by stevenzyang**
* Fixes bug with subsequent uploadFile calls in Xhr.js failing
automatically after the original times out.
* Fixes hanging during logging of part failures by not blocking on it
* Fixes bug with listParts always returning undefined, causing the
client reupload a part it has already uploaded. Remove the byte check in
determining whether or not the file has been uploaded, because it has
shown to be unreliable.
* Removes PART_STATE_COMPUTING_DIGEST state as it is never used in the
code, and is not a useful state to keep track of.

Fixes most of WEBEN-1098, though I have still seen it hang from time to
time.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @stevenzyang has signed the CLA. Thanks for the pull request!

File: src/api/uploads/MultiputPart.js [hunk 1]
```diff
@@ -265,7 +262,7 @@ class MultiputPart extends BaseMultiput {
                 throw new Error('logEvent endpoint not found');
             }
 
-            await retryNumOfTimes(
+            retryNumOfTimes(
```
Comment (mickr) [inline src/api/uploads/MultiputPart.js]: Does this still belong in the `try/catch`?

File: src/api/uploads/MultiputUpload.js [hunk 2]
```diff
@@ -496,13 +495,6 @@ class MultiputUpload extends BaseMultiput {
```
Comment (mickr) [inline src/api/uploads/MultiputUpload.js]: Noticed on L488 a reference to DigestComputing, is that still relevant?

File: src/api/uploads/MultiputPart.js (hunk 1 above)
Comment (stevenzyang) [reply src/api/uploads/MultiputPart.js]: Good point, I'll refactor this to attach a log on error instead

File: src/api/uploads/MultiputUpload.js (hunk 2 above)
Comment (stevenzyang) [reply src/api/uploads/MultiputUpload.js]: Good catch, will remove

**Comment by stevenzyang** [comment]
@mickr Could you take another look?

**Pull request merged**